api/app/sbom.py

```diff
@@ -105,8 +101,8 @@ def _fix_distro(distro: str) -> str:
 
         @staticmethod
         def _find_pkg_mgr(
-            components_map: Dict[str, "TrivyCDXParser.CDXComponent"],
-            refs: List[str],
+            components_map: dict[str, "TrivyCDXParser.CDXComponent"],
+            refs: list[str],
         ) -> Optional["TrivyCDXParser.CDXComponent"]:
```

Comment:
Optional残っているのと、型が文字列になってしまってるので修正が必要そうです
```suggestion
            components_map: dict[str, TrivyCDXParser.CDXComponent],
            refs: list[str],
        ) -> Optional[TrivyCDXParser.CDXComponent]:
```

Reply:
@mshim03 
`Optional["TrivyCDXParser.CDXComponent"]`これをpython3.10の表記方法に変更し
`"TrivyCDXParser.CDXComponent" | None:`とするとエラーが出力されます。

エラー文
`TypeError: unsupported operand type(s) for |: 'str' and 'NoneType'`

Optionalでダブルクオーテーションがあるものをpython3.10の表記方法である`|`で置き換えることに対応していないのでしょうか？
対応していない場合`Optional["TrivyCDXParser.CDXComponent"]`のままでいいでしょうか？

Reply:
まだ定義が完了していない名前を型ヒントとして扱う手法は forward references と呼ばれているそうです。
https://peps.python.org/pep-0484/#forward-references

そして forward references で Unionを表現するときは、 型ヒント全体を文字列にすれば良いそうです
https://docs.python.org/3/library/stdtypes.html#union-type
> Note The | operand cannot be used at runtime to define unions where one or more members is a forward reference. For example, int | "Foo", where "Foo" is a reference to a class not yet defined, will fail at runtime. For unions which include forward references, present the whole expression as a string, e.g. "int | Foo".

今回のケースは以下のようにすればよさそうです
"TrivyCDXParser.CDXComponent | None"

Reply:
教えていただきありがとうございます。
`"TrivyCDXParser.CDXComponent | None"`
で実装したところエラーが出ずにできました。